File: 1.8/Week 5/MVP-Dagger/app/src/main/java/com/example/user/mvp_dagger/view/main/MainActivity.java
package com.example.user.mvp_dagger.view.main;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.user.mvp_dagger.R;
import com.example.user.mvp_dagger.model.StringBean;

import java.util.List;

public class MainActivity extends AppCompatActivity implements MainContract.View {


    private TextView tvMain;
    private EditText etMain;
    private MainPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etMain = findViewById(R.id.etMain);
        tvMain = findViewById(R.id.tvMain);

        presenter = new MainPresenter();
        presenter.attachView(this);

    }

    public void onTextViewUpdate(View view) {

        presenter.updateTextView(etMain.getText().toString());

    }

    @Override
    public void showError(String s) {
        Toast.makeText(this, s, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onUpdateTextView(StringBean stringBean) {
        tvMain.setText(stringBean.getData());

    }

    @Override
    public void onStringAdded(boolean isAdded) {

        String message = null;
        if (isAdded) {
            message = "Added";
        } else message = "Not Added";

        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onListReceived(List<String> strings) {

        Toast.makeText(this, strings.toString(), Toast.LENGTH_SHORT).show();

    }

    public void onStringAdded(View view) {
        presenter.addToList(etMain.getText().toString());
    }

    public void onPrintList(View view) {

        presenter.getStringList();
    }
}

File: 6.25/Week1/ActivityLIfecycleAndIntent/app/src/main/java/com/example/user/activitylifecycleandintent/MainActivity.kt
package com.example.user.activitylifecycleandintent

import android.content.Intent
import android.content.res.Configuration
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView

import com.example.user.activitylifecycleandintent.model.Person
import com.example.user.activitylifecycleandintent.model.PersonP
import com.example.user.activitylifecycleandintent.utils.Constants

class MainActivity : AppCompatActivity() {
    private var etMain: EditText? = null
    private var tvMain: TextView? = null
    private var etPersonName: EditText? = null
    private var etPersonAge: EditText? = null
    private var etPersonGender: EditText? = null


    companion object {
        private val TAG = MainActivity::class.java.simpleName
        private const val TEXTVIEW_KEY = "key"
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "onCreate: ")

        etMain = findViewById(R.id.etMain)
        tvMain = findViewById(R.id.tvMain)
        etPersonName = findViewById(R.id.etPersonName)
        etPersonAge = findViewById(R.id.etPersonAge)
        etPersonGender = findViewById(R.id.etPersonGender)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        Log.d(TAG, "onSaveInstanceState: ")
        //save the data to the bundle
        outState.putString(TEXTVIEW_KEY, tvMain?.text.toString())

    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        Log.d(TAG, "onRestoreInstanceState: ")
        val textViewData = savedInstanceState.getString(TEXTVIEW_KEY)
        tvMain?.text = textViewData
    }


    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        Log.d(TAG, "onConfigurationChanged: ")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart: ")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume: ")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause: ")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop: ")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy: ")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "onRestart: ")
    }

    fun onButtonClicked(view: View) {

        when (view.id) {
            R.id.btnPerson -> {
                //                Create person object
                val person = Person(
                        etPersonName?.text.toString(),
                        etPersonAge?.text.toString(),
                        etPersonGender?.text.toString()
                )

                val personP = PersonP(
                        etPersonName?.text.toString(),
                        etPersonAge?.text.toString(),
                        etPersonGender?.text.toString()
                )
                //                Send person to second activity
                val intent = Intent(applicationContext,
                        SecondActivity::class.java)
                intent.putExtra(Constants.PERSON, person)
                intent.putExtra(Constants.PERSON_P, personP)
                startActivity(intent)
            }
            R.id.btnChangeTextView -> tvMain?.text = etMain?.text.toString()
        }

    }
}
